app/scripts/metamask-controller.js

```diff
@@ -6588,12 +6580,15 @@ export default class MetamaskController extends EventEmitter {
     let rpcPrefs = {};
 
     if (chainId) {
-      const { networkConfigurations } = this.networkController.state;
+      const { networkConfigurationsByChainId } = this.networkController.state;
 
-      const matchingNetworkConfig = Object.values(networkConfigurations).find(
+      const matchingNetworkConfig = Object.values(
+        networkConfigurationsByChainId,
+      ).find(
         (networkConfiguration) => networkConfiguration.chainId === chainId,
       );
 
+      // todo
```

Comment:
Does this need to be resolved as part of this PR?

Reply:
Done in https://github.com/MetaMask/metamask-extension/pull/26433/commits/60b62b9c4972371cab8294dbcb086b61a3352a1b